legacy/reqcounter/reqcounter.go

```diff
@@ -32,6 +32,7 @@ type RequestCounter struct {
 	Since      time.Time     // When the current request counter began recording requests.
 	Interval   time.Duration // The duration of time between each log.
 	Resettable bool          // Reset after logging.
+	Threshold  uint64        // When to warn of a high request count (only for resettable request counters).
```

Comment:
I think you should mention that this threshold is only for a single Interval period.

But also, there's some logical duplication happening here. If Threshold > 0, then obviously we want to log a warning. However, we also control when to log the warning based on whether it is Resettable.

But will it ever make sense to create a RequestCounter that has a Threshold > 0, but where Resettable = false? I think not.

Allowing datatypes to become inconsistent is not good, and should be prevented by either (1) ensuring that the datatype (RequestCounter here) is only able to be created in a guaranteed-correct way, (2) creating a validation check to examine whether the datatype is a good or bad state before using it, or (3) by making it impossible for the datatype to be in a bad state.

I prefer (3) if possible. Here you could remove `Resettable` and just use Threshold to decide whether to warn or not, which would get us to option (3). That is, if Threshold > 0, warn if we exceed it but also reset the counter if we exceed the threshold interval. If Threshold == 0, then never reset the counter and don't bother dealing with the threshold checking logic because it is set to 0.

Reply:
Option 3 is probably the best bet here. I think as long as we document this behavior, it avoids any bad state.